A38-admin-interface-api.md

```diff
@@ -149,7 +147,9 @@ server = ServerBuilder.forPort(50051)
 server.awaitTermination();
 ```
 
-The drawback for this option is dependency management. To date, the gRPC official admin services Channelz/Reflection/Health are distributed as separated package to keep the main package focus as a network library. If we apply this option, then we will void the separate effort.
+This approach is more invasive than the selected solution. Expanding server class increases the complexity of surface API and yet didn't further reduce the minimum LOC needed to create an admin server.
+
+Another drawback for this option is dependency management. To date, the gRPC Channelz service is distributed as a separated package to keep the main package focus as a network library. If we apply this option, then we will void the separate effort. Auto-register pattern doesn't work here, because the add admin service method could be a no-op if users didn't add Channelz as a dependency. Then the documentation needs to explain that the no-op is intended, and the application has to perform following maneuver to add admin services that the application intend to use, which worsen the usability of creating an admin server.
```

Comment:
Sorry, I don't see why this is true.  Dependency management can be handled exactly the same way as in the selected solution.  

Wait -- are you saying that the problem here is that in the selected solution, channelz can be a direct dependency of the `AdminInterface` package, whereas it cannot be a direct dependency of the `ServerBuilder` interface itself?  If so, please state that explicitly.

Regardless of the above, I don't see why it would be a problem for `addAdminServices()` to be a no-op if there are no registered admin services.  I think you can remove that part.

Reply:
Removed this paragraph.

In my view, an API should work to its best extend by default. If a user come across this API in the API reference and try it out, he/she may find the no-op behavior confusing. For an API to work, one needs to update both the code and the dependency, which is less ideal to me.